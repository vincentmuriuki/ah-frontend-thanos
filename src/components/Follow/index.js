import React from 'react';
import PropTypes from 'prop-types';

const Follow = ({ onClick, option }) => (
  <button
    type="button"
    id="bt-follow"
    className={option ? 'btn btn-outline-danger btn-sm m-2' : 'btn btn-outline-primary btn-sm m-2'}
    onClick={() => { onClick(option); }}
  >
    {option ? 'Unfollow' : 'Follow'}
  </button>
);

Follow.propTypes = {
  onClick: PropTypes.func.isRequired,
  option: PropTypes.bool.isRequired,
};

export default Follow;
