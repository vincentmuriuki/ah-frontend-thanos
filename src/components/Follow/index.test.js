import React from 'react';
import { shallow } from 'enzyme';
import Follow from '.';

describe('<Follow /> component', () => {
  it('renders the Follow component', () => {
    const wrapper = shallow(<Follow onClick={jest.fn()} option />);
    expect(wrapper).toMatchSnapshot();
  });
});
