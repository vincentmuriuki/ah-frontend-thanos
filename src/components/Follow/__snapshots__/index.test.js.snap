// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Follow /> component renders the Follow component 1`] = `
<button
  className="btn btn-outline-danger btn-sm m-2"
  id="bt-follow"
  onClick={[Function]}
  type="button"
>
  Unfollow
</button>
`;
